django/db/backends/postgresql/client.py

```diff
@@ -54,8 +55,11 @@ def runshell_db(cls, conn_params):
                     # If the current locale can't encode the data, we let
                     # the user input the password manually.
                     pass
+            default_handler = signal.getsignal(signal.SIGINT)
+            signal.signal(signal.SIGINT, lambda sig, frame: None)
```

Comment:
Good call, I did not know that. However, that necessitated a change in the test. Instead of calling the handler, I'm now just testing that it is in fact set to `SIG_IGN`.